backend/sftp/sftp.go

```diff
@@ -370,6 +370,17 @@ Example:
     umac-64-etm@openssh.com umac-128-etm@openssh.com hmac-sha2-256-etm@openssh.com
 `,
 			Advanced: true,
+		}, {
+			Name:    "host_key_algorithms",
+			Default: fs.CommaSepList{},
+			Help: `Comma separated list of host key algorithms, ordered by preference.
+
+At least one must match with server configuration.
+
+Example:
+
+    ssh-ed25519,ssh-rsa
```

Comment:
I'd probably put all the acceptable values here in the example just to save the users time looking stuff up.

Reply:
First of all, sorry for all the nitpicking (from my part), I really think you made a great first-time contribution! 😃 

In my opinion, after thinking about it, `SpaceSepList` makes more sense here, since other very similar options (?) in this backend such as `key_exchange` and `macs` uses that. Though I do agree that matching the openssh syntax with comma separated does also make sense, but I assume that would be the same with the other options as well? Elsewhere in rclone we seem to use both, e.g. http backend uses comma for headers, hasher backend uses it for hash algorithms.

It seems to me `ssh -Q HostKeyAlgorithms` would given an even more complete set of legal values, including rsa-sha2-256, see: https://security.stackexchange.com/questions/248406/ssh-q-key-not-listing-all-key-types

On the other hand, the resulting list gets long, I agree with that. Maybe we could just make a shorter list which is a realistic example? Seems this is what is done for macs and key_exchange. And these even mention running a corresponding ssh command to see the full list, so could do the same here?

Oh, and doesn't this qualify for `Advanced: true` as well, like the other mentioned related options? This way it will not confuse newbies and will be "hidden" in the rclone config sequence behind:

```
Edit advanced config?
y) Yes
n) No (default)
y/n>
```

Reply:
Not a problem at all - I really appreciate the advice about how to make this as good as it can be.

I agree with the suggestions you have made so I have made those changes and all seems to work well in my testing.

Reply:
Perfect. I like this version. But I'll let @ncw have the final word, I think I sort of hijacked his review...